src/Libraries/Microsoft.Extensions.AI.Abstractions/AudioTranscription/AudioTranscriptionClientExtensions.cs

```diff
@@ -0,0 +1,152 @@
+    /// <returns>The transcriptions generated by the client.</returns>
+    public static IAsyncEnumerable<StreamingAudioTranscriptionUpdate> TranscribeStreamingAsync(
+        this IAudioTranscriptionClient client,
+        Stream audioStream,
+        AudioTranscriptionOptions? options = null,
+        CancellationToken cancellationToken = default)
+        => Throw.IfNull(client)
+            .TranscribeStreamingAsync(
+                [audioStream.ToAsyncEnumerable<DataContent>(mediaType: null, cancellationToken)],
+                options,
+                cancellationToken);
+
+    /// <summary>Transcribes the audio content providing a single audio <see cref="IAsyncEnumerable{DataContent}"/>.</summary>
+    /// <param name="client">The client.</param>
+    /// <param name="audioContent">The single audio content to be transcribed.</param>
+    /// <param name="options">The audio transcription options to configure the request.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>The transcriptions generated by the client.</returns>
+    public static IAsyncEnumerable<StreamingAudioTranscriptionUpdate> TranscribeStreamingAsync(
+        this IAudioTranscriptionClient client,
+        IAsyncEnumerable<DataContent> audioContent,
+        AudioTranscriptionOptions? options = null,
+        CancellationToken cancellationToken = default)
+    {
+        return Throw.IfNull(client)
+            .TranscribeStreamingAsync(
+                [Throw.IfNull(audioContent)],
+                options,
+                cancellationToken);
+    }
+
+    /// <summary>Transcribes the audio content providing a single audio <see cref="DataContent"/>.</summary>
+    /// <param name="client">The client.</param>
+    /// <param name="audioContent">The single audio content to be transcribed.</param>
+    /// <param name="options">The audio transcription options to configure the request.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>The transcriptions generated by the client.</returns>
+    public static IAsyncEnumerable<StreamingAudioTranscriptionUpdate> TranscribeStreamingAsync(
+        this IAudioTranscriptionClient client,
+        DataContent audioContent,
+        AudioTranscriptionOptions? options = null,
+        CancellationToken cancellationToken = default)
+    {
+        IEnumerable<DataContent> audioContents = [Throw.IfNull(audioContent)];
+        return Throw.IfNull(client)
+            .TranscribeStreamingAsync(
+                [audioContents.ToAsyncEnumerableAsync()],
+                options,
+                cancellationToken);
+    }
+
+    private static async IAsyncEnumerable<T> ToAsyncEnumerableAsync<T>(this IEnumerable<T> source)
```

Comment:
This can just be ToAsyncEnumerable.

That's what it's called in the new System.Linq.AsyncEnumerable library. This could presumably just depend on that once it's published.